phoenix-core/src/test/java/org/apache/phoenix/compile/QueryCompilerTest.java

```diff
@@ -2465,10 +2465,10 @@ public void testColumnProjectionUnionAll() throws SQLException {
             assertTrue(rowProj.getColumnProjector(1).getExpression().getMaxLength() == 20);
             assertTrue(rowProj.getColumnProjector(2).getExpression().getDataType()
                 instanceof PVarchar);
-            assertTrue(rowProj.getColumnProjector(2).getExpression().getMaxLength() == 30);
+            assertTrue(rowProj.getColumnProjector(2).getExpression().getMaxLength() == null);
```

Comment:
`/**
     * @return the actual length of the column. For decimal, it would be its precision. For char or
     * varchar, it would be the maximum length as specified during schema definition.
     */
    Integer getMaxLength();

    /**
     * @return scale of a decimal number.
     */
    Integer getScale();
`
I'm not sure about `getMaxLength` and `getScale` return `null` here, it might be better to change the logic to return the actual number instead.  @swaroopak any input here?

Reply:
@yanxinyi Actually, the issue is caused by class [ComparisonExpression](https://github.com/apache/phoenix/blob/e3d3ae5b120189235259280fc074bc7f337aa5b3/phoenix-core/src/main/java/org/apache/phoenix/expression/ComparisonExpression.java#L264). Column type `varchar` union `char(10)` is in compile phase, and it can't determine the maxLength of the result column. API `getMaxLength` and `getScale` may be return `null`, for example, if the input is `varchar` or `decimal` with no limit.
Maybe we should fix the class ComparisonExpression, i will try to look at other possible cases.